
syntax = "proto3";

option csharp_namespace = "XTC.FMP.MOD.Repository.LIB.Proto";
package xtc.fmp.repository;

// 状态
message Status
{
    int32 code = 1;  // 状态码
    string message = 2;  // 状态信息
}

// 空白请求
message BlankRequest
{
}

// 空白回复
message BlankResponse
{
    Status status = 1;  // 状态
}

// 作用域的请求
message ScopeRequest
{
    string scope = 1;  // 作用域
}

// 作用域的回复
message ScopeResponse
{
    Status status = 1;  // 状态
    string scope = 2;  // 作用域
}

// UUID的请求
message UuidRequest
{
    string uuid = 1;  // 唯一识别码
}

// UUID的回复
message UuidResponse
{
    Status status = 1;  // 状态
    string uuid = 2;  // 唯一识别码
}

// 准备上传的回复
message PrepareUploadResponse {
    Status status = 1;  // 状态
    string uuid = 2;  // 唯一识别码
    map<string, string> urls = 3;  // 上传地址列表
}

// 完成上传的回复
message FlushUploadResponse {
    Status status = 1;  // 状态
    string uuid = 2;  // 唯一识别码
    map<string, string> hashs = 3;  // 文件哈希值列表
    map<string, uint64> sizes = 4;  // 文件大小列表
    uint64 flags = 5;  // 标识码
}

// 标志操作的请求
message FlagOperationRequest {
    string uuid = 1;  // 唯一识别码
    uint64 flag = 2;  // 单个标志
}

// 标志操作的回复
message FlagOperationResponse {
    Status status = 1;  // 状态
    string uuid = 2;  // 唯一识别码
    uint64 flags = 3;  // 标志码
}

message FileEntity
{
    string name = 1;  // 文件名
    uint64 size = 2;  // 文件大小
    string hash = 3;  // 文件哈希值
}

message PluginEntity
{
    string uuid = 1;  // 唯一识别码
    string name = 2;  // 名称
    string version = 3;  // 版本号
    uint64 flags = 10;  // 标记
    int64 updatedAt = 11;  // 更新时间
    FileEntity file = 20; // 文件
}

message ModuleEntity
{
    string uuid = 1;  // 唯一识别码
    string org = 2;  // 名称
    string name = 3;  // 名称
    string version = 4;  // 版本号
    string cli = 5; // 框架版本
    uint64 flags = 10;  // 标记
    int64 updatedAt = 11;  // 更新时间
    repeated FileEntity files = 20; // 文件
}

message AgentEntity
{
    string uuid = 1;  // 唯一识别码
    string org = 2;  // 名称
    string name = 3;  // 名称
    string version = 4;  // 版本号
    uint32 port = 5;  // 默认端口号
    repeated string pages = 6;  // 可访问页面列表
    uint64 flags = 10;  // 标记
    int64 updatedAt = 11;  // 更新时间
    FileEntity file = 20; // 文件
}

message ApplicationEntity
{
    string uuid = 1;  // 唯一识别码
    string org = 2;  // 名称
    string name = 3;  // 名称
    string version = 4;  // 版本号
    string platform = 5;  // 运行平台
    uint64 flags = 10;  // 标记
    int64 updatedAt = 11;  // 更新时间
    FileEntity file = 20; // 文件
}
